### PR TITLE
[2157] Document WebDAV Meta endpoint

### LINKED ISSUE #2157: [DAV] resolve file id to user path
Introduced with https://github.com/owncloud/core/pull/35819/files in 10.3

With a PROPFIND to remove.php/dav/meta/${FILEID} and the prop {http://owncloud.org/ns}meta-path-for-user the response will return the path ti the file/folder for the logged in user

### PR DESCRIPTION
**Pull request by settermjd**
This fixes #2157.

### CODE REVIEW

**Comment by settermjd** [comment]
@DeepDiver1975, I don't see a file path in the returned result. Can you review and say what is missing in the examples?

**Comment by DeepDiver1975** [comment]
You are not asking the server for the prop - that's why don't get it ...

{http://owncloud.org/ns}meta-path-for-user has to be in the request and then it will be returned in the response

File: modules/developer_manual/examples/core/webdav_api/meta/report-propfind.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<a:propfind xmlns:a="DAV:" xmlns:oc="http://owncloud.org/ns">
+    <a:prop>
+        <a:getcontentlength />
```
Comment (DeepDiver1975) [inline modules/developer_manual/examples/core/webdav_api/meta/report-propfind.xml]: dont query for props we don't care about

Comment (settermjd) [reply modules/developer_manual/examples/core/webdav_api/meta/report-propfind.xml]: So, all of the properties?

**Comment by settermjd** [comment]
> You are not asking the server for the prop - that's why don't get it ...
> 
> {[http://owncloud.org/ns}meta-path-for-user](http://owncloud.org/ns%7Dmeta-path-for-user) has to be in the request and then it will be returned in the response

And how is that done?

File: modules/developer_manual/examples/core/webdav_api/meta/report-propfind.xml (hunk 1 above)
Comment (settermjd) [reply modules/developer_manual/examples/core/webdav_api/meta/report-propfind.xml]: @DeepDiver1975, which are the essential properties to include?

**Comment by DeepDiver1975** [comment]
> And how is that done?

in the body of the propfind request - this is where the props are requested

File: modules/developer_manual/examples/core/webdav_api/meta/report-propfind.xml (hunk 1 above)
Comment (settermjd) [reply modules/developer_manual/examples/core/webdav_api/meta/report-propfind.xml]: ping @DeepDiver1975

Comment (settermjd) [reply modules/developer_manual/examples/core/webdav_api/meta/report-propfind.xml]: @DeepDiver1975, I removed a number of what seem to be unrequired properties. Can you confirm if these are all the ones to be removed, or if others should be removed as well? Thanks.

**Comment by DeepDiver1975** [comment]
correct request:
```
curl -X PROPFIND http://localhost:8088/remote.php/dav/meta/2147519688 -uadmin:admin --upload-file - -H "Depth: 0" <<end
<?xml version="1.0"?>
<a:propfind xmlns:a="DAV:" xmlns:oc="http://owncloud.org/ns">
<a:prop><oc:meta-path-for-user/></a:prop>
</a:propfind>
end
```

response:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   505    0   354    0   151    255    109 --:--:--  0:00:01 --:--:--   364
<?xml version="1.0"?>
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">
  <d:response>
    <d:href>/remote.php/dav/meta/2147519688/</d:href>
    <d:propstat>
      <d:prop>
        <oc:meta-path-for-user>/foo/bar/foobar.txt</oc:meta-path-for-user>
      </d:prop>
      <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
  </d:response>
</d:multistatus>

**Comment by settermjd** [comment]
@DeepDiver1975, it's ready for another review.

**Comment by DeepDiver1975** [comment]
ah - finally :wink: :+1:

**Comment by mmattel** [comment]
Does this need backporting (10.3 / 10.4) ?

**Comment by phil-davis** [comment]
https://github.com/owncloud/docs/issues/2157
"Introduced with https://github.com/owncloud/core/pull/35819/files in 10.3"

yes - backport to 10.4 and 10.3 needed @settermjd

**Pull request merged**